로그인 API 연결 및 유저 정보 저장

Jo-Myounghee opened the pull request:
## 작업 내용
![image](https://user-images.githubusercontent.com/48939876/142044076-5fd655ea-d3ed-4792-a935-771419c0b297.png)
- 로그인 API 연결
- 유저 정보 전역에 저장

## 고민한 부분
- axios를 썼습니다. 구형 브라우저 지원 및 json 데이터 자동 변환을 통해 편리함을 제공해주기 때문에 적용했습니다. 이에 대해 아침에 이야기 하면 좋을 것 같습니다.
```typescript
function* watchUserInfo() {
  yield takeEvery(USER_INFO_REQUEST, getUserInfo);
}
```
위 함수는 user_info_request 액션이 실행되면 getUserInfo 함수를 실행하는 로직인데, 액션이 실행되었음에도 불구하고 함수가 실행되지 않았습니다. 이를 확인하기 위해 redux-devtools-extension, redux-logger 라이브러리를 설치해서 확인했습니다. 그래서 머지하면 개발 중 콘솔창에 로그가 다 찍힐 텐데, 찍히는 것을 원치 않으시면 src/index.tsx에서 `composeWithDevTools`를 `compose`로 수정해주시면 됩니다.


## 리뷰 포인트
감사합니다.

## 관련된 이슈 넘버
#147 #20

Jo-Myounghee committed:
Feat: CORS 해결을 위한 proxy 설정

react proxy 설정 변경을 통해 cors 에러 해결

#147

Jo-Myounghee committed:
Feat: axios, redux-devtools-extension, redux-logger 라이브러리 설치

구형 브라우저 지원 및 json 데이터 자동 변환을 통해 편리함을 제공하여 axios 일단 적용
redux-saga를 사용하면서 로그를 확인하기 위해 2개의 라이브러리 설치

#147

Jo-Myounghee committed:
Feat: redux-logger 적용

콘솔창에서 redux 상태를 확인하기 위해 redux-logger 적용

#147

Jo-Myounghee committed:
Feat: user 정보 받아오는 로직

로그인 요청 후 서버측에서 성공, 실패 여부를 보내주고, 성공 시에는 user data를 함께 넘겨주기 때문에 성공 시 바로 redux에 user 정보를 담아주게 구현

#147

Jo-Myounghee committed:
Feat: 우측 상단 모달에 유저 닉네임 및 프로필 사진 추가

로그인이 완료되면 받는 유저 정보를 사용해 유저 닉네임, 프로필 사진 추가

#20

Jo-Myounghee committed:
Feat: 로그인 버튼 활성화

/login url로 접근했을 때 존재하는 로그인 버튼 활성화

#147

ddb8036631 commented on frontend/package.json:
```diff
@@ -71,5 +71,6 @@
     "prettier-eslint-cli": "^5.0.1"
-  }
+  },
+  "proxy": "http://localhost:5000"
```
`package.json`에 작성된 프록시는 나중에 배포된 서버로 연결이 쉽지 않을 것 같아 보입니다. 나중에 `.env`로 개발과 배포 환경에서의 URL을 따로 작성한 뒤, 밑에 `getGroupInfoApi`함수에서의 URL은 앞에 `.env`에서 읽어들이면 어떨까요 ? 🧐

Jo-Myounghee replied:
그렇게하면 프론트에서 바로 요청보내는 거라 CORS에러가 나더라구요. 
해결을 위해서 package.json에서 proxy 처리를 했는데, 배포된 서버도 생각하면 proxy를 여러 개 설정해서 분기해도 될 거 같습니다 !

ddb8036631 reviewed (approved):
saga 구조 짜느라 고생하셨습니다👍👍

gofeel8 commented on frontend/src/Router.tsx:
```diff
@@ -1,15 +1,24 @@
+  const dispatch = useDispatch();
+  const { user } = useSelector((state: RootState) => state.user);
+
+  const getUserInfo = useEffect(() => {
```
getUserInfo 변수에 할당해준 이유가 있을까요

Jo-Myounghee replied:
아뉘여 딱히 이유는 없습니다 빼도 될 거 같아여 좀따 수정 하겠습니다~!

soosungp33 reviewed (approved):
saga 봐도봐도 어렵네요..고생하셨어요!

gofeel8 reviewed (approved):
redux-saga 코드 잘 보구갑니다~! 쉽지 않은데 먼저 배우고 도입하느라 고생하셨습니다~!

Jo-Myounghee committed:
Fix: merge conflict 해결

# Conflicts:
#	frontend/src/Router.tsx
#	frontend/src/components/Header/Profile/index.tsx
#	frontend/src/index.tsx
